### PR TITLE
added details to the tutorial and changed some content of the README DEV-154

### PR DESCRIPTION
**Pull request by jenperson**
Changed some of the steps of the README so that all of the Squid steps are in the tutorial instead of having some of its setup in the README
Added step numbers, details, and links to the tutorial. Changed some wording and added a summary step with a CTA.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,40 +1,15 @@
...
-```bash
-cd todo/backend
-```
-
-4. Install the required packages:
+1. The frontend of this project is a React app that uses [Vite](https://vitejs.dev/guide/cli.html). Note that Squid works with any TypeScript client; this project just happens to use React with Vite. Intall the Vite CLI:
 
 ```bash
-npm install
+npm install -g vite
```
Comment (ohsnapitscolin) [inline README.md]: @jenperson you shouldn't need to do this step, `vite` should already be installed if you run `npm install` in the `frontend` directory.

File: frontend/public/tutorial.md [hunk 2]
```diff
@@ -226,25 +336,43 @@ export class ExampleService extends SquidService {
 }
 ```
 
-In the output of the locally running backend, you should see a log like:
+2. Upon saving the file, in the output of the locally running backend, a log appers with a format similar to the following:
```
Comment (ohsnapitscolin) [inline frontend/public/tutorial.md]:
```suggestion
2. Upon saving the file, in the output of the locally running backend, a log appears with a format similar to the following:
```

**Review by ohsnapitscolin — COMMENTED** [review]
Looks great overall! Just left a few small comments.

File: README.md [hunk 3]
```diff
@@ -1,40 +1,15 @@
...
-```
-
-6. Start the backend in this terminal window by running the following command:
-
-```bash
-squid start
-```
-
-7. Open a second terminal window. In this window, navigate to the frontend:
+2. Open a terminal window, and then navigate to the frontend:
```
Comment (etaibar3) [inline README.md]: The step numbers need to be fixed (now skip from 2 to 8)

File: frontend/public/tutorial.md [hunk 4]
```diff
@@ -1,12 +1,64 @@
...
-- In `main.tsx`, update the options passed `SquidContextProvider` with you `appId` and `squidDeveloperId`.
-- Run the `Create .env file` script in (found in the [Squid Console](https://console.squid.cloud)) in the `backend` directory.
-- Run `squid start` in the backend directory.
+Welcome to your first Squid project! Through a few simple steps, you will add functionality to a full-stack to-do application and learn about some of the features of Squid.
 
-## Creating a Todo
+1. In the [Squid Cloud Console](https://console.squid.cloud) create a new app called `todo`.
 
-Querying for todos. In `App.tsx` add the following:
+2. To initialize Squid in a client app, you must provide some details about your Squid application. In the `main.tsx` file of the frontend, update the options passed to `SquidContextProvider` with your `appId` and `squidDeveloperId`. You can find these values in the [Squid Cloud Console](https://console.squid.cloud):
```
Comment (etaibar3) [inline frontend/public/tutorial.md]: I'd maybe specify "You can find these values in the **application overview page** of the Squid Cloud Console"

File: frontend/public/tutorial.md [hunk 5]
```diff
@@ -1,12 +1,64 @@
...
+      appId: 'YOUR_APP_ID', // Update the App ID
+      region: 'us-east-1.aws',
+      environmentId: 'dev',
+      squidDeveloperId: 'YOUR_SQUID_DEVELOPER_ID', // Update the developer ID
+    }}
+  >
+...  
+```
+
+3. In the [Squid Cloud Console](https://console.squid.cloud), scroll to the **Backend project** section and click **Create .env vars**.
```
Comment (etaibar3) [inline frontend/public/tutorial.md]: The button is "Create .env **file**"

File: frontend/public/tutorial.md [hunk 6]
```diff
@@ -1,12 +1,64 @@
...
+
+The Squid CLI lets you run your Squid projects locally during development and testing, and then deploy your backend when ready.
+
+5. Navigate to the `backend` directory:
+
+```bash
+cd backend
+```
+
+6. Copy the command to create the `.env` file that's shown in the **Backend project** section of the Squid Cloud Console, and then run it in the `backend` directory. The script has the following format:
```
Comment (etaibar3) [inline frontend/public/tutorial.md]: nit: We say "command" and then say "script". I'd just stick to one term in an effort to eliminate any possible confusion

File: frontend/public/tutorial.md [hunk 7]
```diff
@@ -1,12 +1,64 @@
...
+
+```bash
+squid start
+```
+
+Your Squid backend is now running locally for development and testing! Click **Next** to continue to the next step where you add your fist Squid functionality to this app.
+
+## Creating a To-do
+
+Right now, the **Add Todo** button displays a modal for adding a new to-do to the list, but currently, saving a to-do does not produce any effect. Let's add some code that writes new to-do tasks to Squid's [built-in database](https://docs.squid.cloud/docs/integrations/database/built-in) and also listens for updates to the to-dos.
```
Comment (etaibar3) [inline frontend/public/tutorial.md]: nit: this first sentence seems wordy/run-on-ish to me (i.e. the repeated "Right now" and "currently) (but my grammar could also be wrong)

File: frontend/public/tutorial.md [hunk 8]
```diff
@@ -43,11 +95,11 @@ const App = () => {
...
 ```
 
-Now test out the "Add Todo" button!
+Now test out the **Add Todo** button! Notice as you add new to-dos, they appear in the view in real-time. To learn how to update a to-do, click **Next**.
 
-## Updating a Todo
+## Updating a To-do
 
-In `App.tsx`
+1. To update the contents of an existing to-do, we use the `update` method of the [Squid Client SDK](https://docs.squid.cloud/docs/development-tools/client-sdk/mutations#update). In `App.tsx`, update the `handleToggle` function with the following code:
```
Comment (etaibar3) [inline frontend/public/tutorial.md]: Unless I misunderstand the functionality, we aren't actually updating the content but rather updating its status. Not sure if content update is WIP though

**Review by etaibar3 — APPROVED** [review]
LGTM!

File: frontend/public/tutorial.md [hunk 9]
```diff
@@ -1,12 +1,66 @@
...
+      environmentId: 'dev',
+      squidDeveloperId: 'YOUR_SQUID_DEVELOPER_ID', // Update the developer ID
+    }}
+  >
+...  
+```
+
+3. In the [Squid Cloud Console](https://console.squid.cloud), scroll to the **Backend project** section and click **Create .env file**. 
+
+<img src="./copy_env.gif" alt="drawing" width="500"/>
```
Comment (ohsnapitscolin) [inline frontend/public/tutorial.md]: @jenperson I'd remove this for now, since we don't have image support yet.

**Review by ohsnapitscolin — APPROVED** [review]

**Pull request merged**